penaltymodel_lp/penaltymodel/lp/generation.py

```diff
@@ -160,10 +162,17 @@ def generate_bqm(graph, table, decision_variables,
     cost_weights[0, -1] = -1     # Only interested in maximizing the gap
 
     # Returns a Scipy OptimizeResult
-    result = linprog(cost_weights.flatten(), A_eq=noted_matrix, b_eq=noted_bound,
-                     A_ub=unnoted_matrix, b_ub=unnoted_bound, bounds=bounds)
-
-    #TODO: propagate scipy.optimize.linprog's error message?
+    # Note: if linear program encounters an ill conditioned matrix or non-full-row-rank matrix,
+    #   rather than let it continue, just fail and allow the next penaltymodel to make an attempt
```

Comment:
this should probably say why we are "failing" on a warning case where (as far as I know) it would still work, just emit a warning that we could otherwise suppress.

Reply:
I don't quite understand what you are suggesting. Are you saying that I should just emit the warning? Or are you saying that I should elaborate on why I am forcing it to fail?

Reply:
If I understand correctly what is happening, the warning that is emitted by `linprog` is just telling the user that it is less efficient that it could be, _not_ that it will give an incorrect answer. In this, you are doing two things, you are suppressing the warning and then raising an exception. I understand (from the comment/code) why you are suppressing the warning but not why you are then raising an exception

Reply:
If the problem matrix is ill-conditioned, the solution found is not reliable. (i.e. minute changes to the problem matrix will lead to a significantly different solution). Hence, if these warnings come up, I would just pass it to MIP via an exception.

The non-full-row-rank matrix warning could be okay. This would likely just be an efficiency thing.

Reply:
Ah, there we go, the first sentence is the one I was missing. Perhaps we should have the behaviour switch on those two warnings then?

Reply:
I've updated the wording. Please let me know if it's okay. 

Note that I would like to have a nicer fix for this in the coming weeks. Right now, it's just meant to be a reliable, quick fix.